sdk/log/simple.go

```diff
@@ -55,7 +94,22 @@ func (s *SimpleProcessor) OnEmit(ctx context.Context, r *Record) error {
 		simpleProcRecordsPool.Put(records)
 	}()
 
-	return s.exporter.Export(ctx, *records)
+	err := s.exporter.Export(ctx, *records)
+
+	if s.selfObservabilityEnabled {
+		if err != nil {
+			s.processedMetric.Add(context.Background(), int64(len(*records)),
+				s.processedMetric.AttrComponentType(otelconv.ComponentTypeSimpleLogProcessor),
+				s.processedMetric.AttrComponentName(s.componentName),
+				s.processedMetric.AttrErrorType(otelconv.ErrorTypeOther))
+		} else {
+			s.processedMetric.Add(context.Background(), int64(len(*records)),
+				s.processedMetric.AttrComponentType(otelconv.ComponentTypeSimpleLogProcessor),
+				s.processedMetric.AttrComponentName(s.componentName))
+		}
```

Comment:
This section contains some duplicate code; can we consolidate it?

Reply:
Done!